clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -2403,17 +2404,46 @@ private ClusterResourceListeners configureClusterResourceListeners(Deserializer<
         return clusterResourceListeners;
     }
 
-    private void close(long timeoutMs, boolean swallowException) {
+    private Timer createTimerForRequest(final Duration timeout) {
+        final Time localTime = (time == null) ? new SystemTime() : time;
```

Comment:
mind elaborate on when the time object can be null?  maybe when we inject the time object?

Reply:
The `close()` could be called [from the constructor](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java#L827) of this class itself when an exception is thrown before the field `this.time` could be initialized. I can avoid it by setting the field as the first thing that happens in the constructor but then we could be creating a coupling with the initialization order of fields in the constructor to the close method. Instead, to be on the safer side, a better approach IMO is to handle the null case explicitly. 

I have added a comment explaining when it could be null in the upcoming commit.

Reply:
How about Objects.requireNotNull(time) in the constructor? So that we could avoid the null handling.